File: src/ru/hudspeed/pro/GpsCarAppService.java
package ru.gpsantiradar.app;

import androidx.car.app.CarAppService;
import androidx.car.app.Session;
import androidx.car.app.validation.HostValidator;

public final class GpsCarAppService extends CarAppService {
    @Override public HostValidator createHostValidator() {
        return new HostValidator.Builder(this).build();
    }

    @Override public Session onCreateSession() {
        return new GpsCarSession();
    }
}

File: src/main/kotlin/ru/gpsantiradar/app/GpsCarAppService.kt
package ru.gpsantiradar.app

import android.annotation.SuppressLint
import androidx.car.app.CarAppService
import androidx.car.app.Session
import androidx.car.app.validation.HostValidator

class GpsCarAppService : CarAppService() {
    @SuppressLint("PrivateResource")
    override fun createHostValidator(): HostValidator = HostValidator.Builder(this)
        .addAllowedHosts(androidx.car.app.R.array.hosts_allowlist_sample).build()
    override fun onCreateSession(): Session = GpsCarSession()
}
